panels/background/cc-background-xml.c

```diff
@@ -426,6 +431,33 @@ cc_background_xml_load_from_dir (const gchar      *path,
   g_object_unref (enumerator);
 }
 
+static gchar *
+read_vendor_background_directory (void)
+{
+  g_autoptr(GKeyFile) keyfile = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree gchar *result = NULL;
+
+  keyfile = g_key_file_new ();
+  if (!g_key_file_load_from_file (keyfile, VENDOR_CONF_FILE, G_KEY_FILE_NONE, &error)) {
+    if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+      g_warning ("Could not read file %s: %s", VENDOR_CONF_FILE, error->message);
+
+    return NULL;
+  }
+
+  result = g_key_file_get_string (keyfile,
+                                  VENDOR_BACKGROUND_GROUP,
+                                  VENDOR_BACKGROUND_DIRECTORY_KEY,
+                                  &error);
+  if (!result) {
+    g_warning ("Could not read backgrounds path from %s: %s", VENDOR_CONF_FILE, error->message);
```

Comment:
On second thought, I think we should avoid printing this warning when the group or key are not found. Something like what we did here: https://github.com/endlessm/gnome-control-center/blob/5fe3074928bfe07fe33790d900386c2d6529c6aa/panels/user-accounts/um-photo-dialog.c#L410

Reply:
We should move to a common function eventually.